packages/dds/tree/docs/user-facing/schema-evolution.md

```diff
@@ -0,0 +1,304 @@
+# Schema Evolution
+
+This document outlines the general constraints around maintaining compatibility in an application document ecosystem in the face of schema changes.
+
+## Context
+
+When an application creates and uses a schema to open a document using `SchemaFactory` and `TreeView.initialize`,
+information about that schema is persisted as part of the document.
+The schema that the application works with in code is hereforth referred to as "view schema" and the schema persisted with the document "stored schema."
+
+Compatibility scenarios can be described in terms of discrepancies between the view schema and the stored schema.
+For example, an application in the geometry domain might try to open a document using a view schema constructed with:
+
+```typescript
+const builder = new SchemaFactory("Geometry");
+
+class Point extends builder.object({
+	x: builder.number,
+	y: builder.number,
+}) {}
+
+class Circle extends builder.object({
+	center: Point,
+	radius: builder.number,
+}) {}
+
+class Plane extends builder.object({
+	shapes: builder.array([Circle, Point]),
+}) {}
+```
+
+but discover that the document's stored schema doesn't allow `Point` children under the `shapes` field, i.e. its definition of `Plane` was this:
+
+```typescript
+class Plane extends builder.object({
+	shapes: builder.array([Circle]),
+}) {}
+```
+
+Applications may leverage `SharedTree`'s compatibility APIs to implement policies that allow clients running incompatible code to fail in predictable ways.
+
+## General Approach
+
+SharedTree allows application authors to make _backward-compatible changes to document schema_ using `TreeView.upgradeSchema`.
+A change to document schema is backward-compatible if the set of documents allowed by the old schema is a subset of documents allowed by the new schema.
+Put differently, valid documents must remain valid.
+
+The following types of changes are backward-compatible under this definition:
+
+1. Adding an optional field to an object node
+2. Adding to the set of allowed types for a field
+3. Relaxing a field kind to a more general field kind
+
+Adding new schemas (rather than modifying existing ones) is allowed and a typical reason to make a change to the allowed types for a field.
```

Comment:
Nit (I had to reread in order to parse correctly)
```suggestion
Adding new schemas (rather than modifying existing ones) is allowed and is a typical reason to make a change to the allowed types for a field.
```